Introductory Tests/udp_initial_condition.py

```diff
@@ -78,71 +90,54 @@ def compute_trajectory(self, x):
 
         Returns:
             fintess_values: Evaluation of proximity of satelite to target altitude.
-            r_store:        Array containing values on position at each time step for the trajectory.
-            v_store:        Array containing values on velocities at each time step for the trajectory.
-            a_store:        Array containing values on acceleration at each time step for the trajectory.
+            trajectory_info: Numpy array containing information on position and velocity at every time step.
         """
-        # Initial information
-        r = np.transpose(x[0:3]) # Start Position
-        v = np.transpose(x[3:6]) # Initial velocity
 
-        # Array containing times for summation
-        time_list = np.arange(self.start_time, self.final_time, self.time_step)
+        # Fitness value (to be maximized)
+        fitness_value = 0
 
-        # Numpy Arrays to store trajectory information
-        r_store = np.zeros((3,len(time_list)))
-        v_store = np.zeros((3,len(time_list)))
-        a_store = np.zeros((3,len(time_list)))
+        # Setup algorithm of choice
+        intg = integrator(self.body_mesh, self.mesh_vertices, self.mesh_faces, self.body_density, self.target_altitude, self.final_time, self.start_time, self.time_step, self.algorithm)
 
-        # Add starting position to memory
-        r_store[:,0] = r
-        v_store[:,0] = v
+        # Integrate trajectory
+        trajectory_info = intg.run_integration(x)
 
-        # Fitness value (to be maximized)
-        fitness_value = 0
+        #D.set_float_fmt('float64')
+        #initial_state = D.array(x)
+        #a = de.OdeSystem(self.equation_of_motion, y0=initial_state, dense_output=True, t=(self.start_time, self.final_time), dt=self.time_step, rtol=1e-12, atol=1e-12)
+        #a.method = "RK87"
+        #a.integrate()
+        #trajectory_info = np.transpose(a.y)
         
-        # Numerical integration of Newton's equations of motion (trajectory propagation)
-        r_store, v_store, a_store = self.euler_approx(r, v, time_list, r_store, v_store, a_store)
-
         # Return fitness value for the computed trajectory
-        squared_altitudes = r_store[0,:]**2 + r_store[1,:]**2 + r_store[2,:]**2
+        squared_altitudes = trajectory_info[0,:]**2 + trajectory_info[1,:]**2 + trajectory_info[2,:]**2
         fitness_value = np.mean(np.abs(squared_altitudes-self.target_altitude))
-        return fitness_value, r_store, v_store, a_store
+        return fitness_value, trajectory_info 
+
+
 
-    
-    def euler_approx(self, r, v, time_list, r_store, v_store, a_store):
-        """euler_approx uses euler's method as numerical integrator for approximating the trajectory. 
+
+
+    # Used by all RK-type algorithms
+    def equation_of_motion(self, _, x):
```

Comment:
duplicate, could just define in a separate file. Possibly to get rid of passing the mesh around etc, you could make a tiny class that holds the mesh with this function and just pass that around? Would also solve the discussed problem of the mesh being inside the integrator